Source/DafnyLanguageServer.Test/Lookup/ReferencesTest.cs

```diff
@@ -0,0 +1,278 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Dafny.LanguageServer.IntegrationTest.Extensions;
+using Microsoft.Dafny.LanguageServer.IntegrationTest.Util;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Progress;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Lookup {
+  public class ReferencesTest : ClientBasedLanguageServerTest {
+    private IRequestProgressObservable<IEnumerable<Location>, LocationContainer> RequestReferences(
+      TextDocumentItem documentItem, Position position) {
+      return client.RequestReferences(
+        new ReferenceParams {
+          TextDocument = documentItem.Uri,
+          Position = position
+        }, CancellationToken);
+    }
+
+    /// <summary>
+    /// Assert that when finding-references at each cursor position,
+    /// the client returns all ranges marked with regular spans.
+    /// </summary>
+    /// <param name="source"></param>
+    private async Task AssertReferences(string source) {
+      MarkupTestFile.GetPositionsAndRanges(
+        source, out var cleanSource, out var positions, out var expectedRangesArray);
+      var expectedRanges = new HashSet<Range>(expectedRangesArray);
```

Comment:
Seems like you could extract positions using `expectedRangesArray.Select(r => r.Start)[.Concat(expectedRangesArray.Select(r => r.End))]` to replace many of `><` tokens you have in the tests.

Reply:
Good idea, I've implemented something along those lines to simplify the tests' markup.